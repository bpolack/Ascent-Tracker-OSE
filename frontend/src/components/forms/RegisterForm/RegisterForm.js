import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setAlert } from '../../../actions/alert';
import PropTypes from 'prop-types';

// Bootstrap Components
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';

export class RegisterForm extends Component {
	constructor(props) {
		super(props);
		this.handleChange = this.handleChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleChange(event) {
		event.preventDefault();
		this.setState({
			[event.target.name]: event.target.value,
		});
	}

	handleSubmit(event) {
		event.preventDefault();
		if (this.state.password !== this.state.password2) {
			this.props.setAlert('Passwords do not match', 'error');
		}
		else {

		}
	}

	render() {
		return (
			<Form onSubmit={this.handleSubmit}>
				<Form.Group controlId="registerFirstName">
					<Form.Label>First Name</Form.Label>
					<Form.Control type="text" name="firstName" onChange={this.handleChange} required />
				</Form.Group>
				<Form.Group controlId="registerLastName">
					<Form.Label>Last Name</Form.Label>
					<Form.Control type="text" name="lastName" onChange={this.handleChange} required />
				</Form.Group>
				<Form.Group controlId="registerEmail">
					<Form.Label>Email Address</Form.Label>
					<Form.Control type="email" name="email" onChange={this.handleChange} required />
				</Form.Group>
				<Form.Group controlId="registerPassword">
					<Form.Label>Password</Form.Label>
					<Form.Control type="password" name="password" onChange={this.handleChange} required />
					<Form.Text id="passwordHelpBlock" muted>
						Password must be at least 8 characters long, and contain at least one number or symbol.
					</Form.Text>
				</Form.Group>
				<Form.Group controlId="registerPassword2">
					<Form.Label>Confirm Password</Form.Label>
					<Form.Control type="password" name="password2" onChange={this.handleChange} required />
					<Form.Text id="passwordHelpBlock" muted>
						Confirm your password, by entering it a second time
					</Form.Text>
				</Form.Group>
				<Button className="px-4" variant="primary" type="submit">Sign Up</Button>
			</Form>
		)
	}
}

RegisterForm.propTypes = {
	setAlert: PropTypes.func.isRequired
}

export default connect(null, { setAlert })(RegisterForm)
