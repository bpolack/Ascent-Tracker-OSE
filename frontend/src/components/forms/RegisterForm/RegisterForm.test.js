import React from 'react';
import { shallow } from 'enzyme';
import { RegisterForm } from './RegisterForm';

describe('RegisterForm', () => {
	it("Renders correctly without crashing", () => {
		const registerMock = jest.fn();

		const wrapper = shallow(<RegisterForm isAuthenticated={false} register={registerMock} />);
		expect(wrapper).toMatchSnapshot();
	});

	describe('When user changes input', () => {
		it('form state is updated', () => {
			const registerMock = jest.fn();
			const preventDefault = jest.fn();

			const wrapper = shallow(<RegisterForm isAuthenticated={false} register={registerMock} />);

			// Change value of input field
			const fnameElement = wrapper.find('input[name="fname"]');
			const lnameElement = wrapper.find('input[name="lname"]');
			const emailElement = wrapper.find('input[name="email"]');
			const passElement = wrapper.find('input[name="password"]');
			fnameElement.simulate('change', { preventDefault, target: { name: 'fname', value: 'John' } });
			lnameElement.simulate('change', { preventDefault, target: { name: 'lname', value: 'Smith' } });
			emailElement.simulate('change', { preventDefault, target: { name: 'email', value: 'testing@testing.com' } });
			passElement.simulate('change', { preventDefault, target: { name: 'password', value: 'Access1234!!!' } });
			
			// Check that state has updated to match
			expect(wrapper.state('fname')).toEqual("John");
			expect(wrapper.state('lname')).toEqual("Smith");
			expect(wrapper.state('email')).toEqual("testing@testing.com");
			expect(wrapper.state('password')).toEqual("Access1234!!!");
		});
	});

	describe('When user submits the form', () => {
		it('details are passed to register function', () => {
			const registerMock = jest.fn();
			const preventDefault = jest.fn();

			const wrapper = shallow(<RegisterForm isAuthenticated={false} register={registerMock} />);

			// Set values and simulate submit
			const formElement = wrapper.find('form');
			const fnameElement = wrapper.find('input[name="fname"]');
			const lnameElement = wrapper.find('input[name="lname"]');
			const emailElement = wrapper.find('input[name="email"]');
			const passElement = wrapper.find('input[name="password"]');
			fnameElement.simulate('change', { preventDefault, target: { name: 'fname', value: 'John' } });
			lnameElement.simulate('change', { preventDefault, target: { name: 'lname', value: 'Smith' } });
			emailElement.simulate('change', { preventDefault, target: { name: 'email', value: 'testing@testing.com' } });
			passElement.simulate('change', { preventDefault, target: { name: 'password', value: 'Access1234!!!' } });
			formElement.simulate('submit', { preventDefault });

			// Check that mock function called & with correct params
			expect(registerMock).toHaveBeenCalledTimes(1);
			expect(registerMock).toHaveBeenCalledWith({"fname": "John", "lname": "Smith", "email": "testing@testing.com", "password": "Access1234!!!"});
		});
	});
});