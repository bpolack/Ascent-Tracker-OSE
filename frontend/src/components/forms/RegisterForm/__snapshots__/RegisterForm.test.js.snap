// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm Renders correctly without crashing 1`] = `ShallowWrapper {}`;
